sematic/ui/src/pipelines/RunPanel.tsx

```diff
@@ -1,30 +1,92 @@
-import { Box } from "@mui/material";
+import { Box, styled } from "@mui/material";
 import { FlowWithProvider } from "./graph/ReactFlowDag";
 import { usePipelinePanelsContext } from "../hooks/pipelineHooks";
 import { useGraphContext } from "../hooks/graphHooks";
 import { RunDetailsPanel } from "./RunDetailsPanel";
+import RunPanelContext from "./RunDetailsContext";
+import { useMemo, useRef, useState } from "react";
+import { ExtractContextType } from "../components/utils/typings";
+import Loading from "../components/Loading";
+
+const FloatingFooter = styled('div')`
+  width: 100%;
+  position: sticky;
+  bottom: 0;
+  height: 0;
+`;
+
+const FloatingFooterAnchor = styled('div')`
+  width: 100%;
+  position: absolute;
+  bottom: 0;
+`;
+
+const LoadingOverlay = styled(Box)`
+  width: 100%;
+  position: absolute;
+  z-index: 251;
+  height: 100%;
+  pointer-events: none;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`
 
 export default function RunPanel() {
   const { graph } = useGraphContext();
   const { selectedPanelItem } = usePipelinePanelsContext();
 
+  const scrollerId = 'run-panel-scrolling-area';
+  const scrollContainerRef = useRef<HTMLElement>();
+
+
+  const [footerRenderProp, setFooterRenderPropState] = useState<(() => JSX.Element) | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const runDetailsContextValue = useMemo<
+        ExtractContextType<typeof RunPanelContext>
+    >(() => ({
+      setFooterRenderProp: (renderProp) => {
+        setFooterRenderPropState(() => renderProp);
+      },
+      scrollerId,
+      scrollContainerRef,
+      setIsLoading
+    }), [setFooterRenderPropState]);
+
   if (!graph) {
     return <></>;
   }
   return (
-    <Box sx={{ gridColumn: 2, gridRow: 2, overflowY: "scroll" }}>
-      {selectedPanelItem === "graph" && (
-        <>
-          <FlowWithProvider
-            // Nasty hack to make sure the DAG is remounted each time to trigger a ReactFlow onInit
-            // to trigger a new layout
-            key={Math.random().toString()}
-          />
-        </>
-      )}
-      {selectedPanelItem === "run" && (
-          <RunDetailsPanel />
-      )}
+    <Box sx={{ gridColumn: 2, gridRow: 2, height: '100%', overflow: 'hidden', 
+    position: 'relative'}}>
+        {isLoading && <LoadingOverlay>
+          <Loading isLoaded={false} />
+        </LoadingOverlay>}
+      <Box id={scrollerId} ref={scrollContainerRef}
+      sx={{ overflowY: "scroll", position: 'relative', height: '100%' }}>
+        <RunPanelContext.Provider value={runDetailsContextValue}>
+
+          {selectedPanelItem === "graph" && (
+            <>
+              <FlowWithProvider
+                // Nasty hack to make sure the DAG is remounted each time to trigger a ReactFlow onInit
+                // to trigger a new layout
+                key={Math.random().toString()}
```

Comment:
Do you have a long-term solution?

Reply:
Thanks for asking about it.  So far, despite the section shown as added code, I was simplifying moving an existing code section to here. I didn't write the code, so I will have to investigate further to know what this is about. Later we will rework the DAG section altogether. I will take a closer look then.